### ISSUE TITLE
Failed to apply binary diff

### ISSUE THREAD

**Issue by mstanic0** [bitrise-steplib/steps-git-clone#127]
Git clone step is failing for me on version 4.x.x while on version 3.x.x it works.

The error I see in the console is the following:
```
$ git "apply" "--index" "/tmp/248.diff167610725"
error: cannot apply binary patch to 'shake/src/main/res/drawable-mdpi/shake_sdk_private_view_pattern.png' without full index line
```

**Comment by bitce** [comment]
Hi @mstanic0!

Can you send us an URL for an example build with both situations? Thanks!

**Comment by mstanic0** [comment]
Failed: https://app.bitrise.io/build/c1491dd5cd0319d7#?tab=log
Success:

**Comment by bitce** [comment]
Thanks! Could you perhaps enable Support User access for this app? Could help provide further info as I'm not able to recreate it so far.

**Comment by mstanic0** [comment]
I have enabled Support User access for the app.

It fails on the git clone step when I change the raw file in my project eg. image.
When I downgrade the git clone step to the older version it works then.

**Comment by ofalvai** [comment]
Hi @mstanic0,

We confirmed that this is indeed a bug in the git clone step, we appreciate the bugreport. A new step release will fix this, in the meantime you can use the downgraded version of the step.

**Comment by mstanic0** [comment]
Thanks!

**Comment by istvankovacs-bitrise** [comment]
Hi @mstanic0,  
  
We will investigate the possibility of rentroducing the retry mechanism for the diff file checkout method since the lack of it causes the difference between the two pipelines you referenced.

In the meantime, you may set the `ManualMerge` option of the `git-clone` Step to false, in order to force a recursive merging strategy, which should be able to merge in an expected manner.

**Comment by istvankovacs-bitrise** [comment]
Hi @mstanic0, we have released v4.1.0 last week, where we reintroduced a fallback option, in case of an error has occurred during applying a patch.

However, the root cause of this issue is caused by a malfunctioning endpoint of the _Bitbucket_ API. We have reported it to _Atlassian_, and they acknowledged and confirmed that it is indeed an issue on their side. You may follow the issue ticket [here](https://jira.atlassian.com/browse/BCLOUD-20896).
